lib/ohai/plugins/cpu.rb

```diff
@@ -46,13 +47,211 @@ def parse_bsd_dmesg(&block)
     cpuinfo
   end
 
-  collect_data(:linux) do
+  def range_to_a(range)
+    range_array = []
+    range.split(",").each do |cpu|
+      if /\d+-\d+/.match?(cpu.to_s)
+        range_array << Range.new(*cpu.split("-").map(&:to_i)).to_a
+      else
+        range_array << cpu.to_i
+      end
+    end
+    range_array.flatten
+  end
+
+  def parse_lscpu(cpu_info)
+    lscpu_info = Mash.new
+    lscpu_info[:numa_node_cpus] = Mash.new
+    lscpu_info[:vulnerability] = Mash.new
+    begin
+      so = shell_out("lscpu")
+      cpu_cores = shell_out("lscpu -p=CPU,CORE,SOCKET")
+      if so.exitstatus == 0 && cpu_cores.exitstatus == 0
```

Comment:
can remove this and just `rescue` (anything, including ShellCommandFailed)

Reply:
I agree, @ramereth  but I'll see if there's something @lamont-granquist saw that I"m missing